### ISSUE TITLE
Bad credentials suddenly

### ISSUE THREAD

**Issue by Reckziegel182** [dubocr/homebridge-tahoma#204]
Hi there, 

all of a sudden, i am getting this error
![image](https://user-images.githubusercontent.com/72069742/94546921-d50c7480-024e-11eb-958b-e577c4db84fc.png)

Looks like my ip has been blacklisted, when i use this link https://www.tahomalink.com/enduser-mobile-web/steer-html5-client/tahoma/ i get bad username and password, but when i connect using the LTE from my mobile, i get in. Any way to get around this please?

thank you

Martin

**Comment by jasper-seinhorst** [comment]
I'm having the same issue. I think the IP blacklist should be a matter of time because I've seen the error in the past. I also noticed that due to this error my Homebridge isn't starting anymore. Had to remove the plugin from the config. Looks like Tahoma is changing things at their end, since a week my Awnings have incorrect states, and now this issue occurs.

@dubocr do you have a clue what is going on?

**Comment by vdheidenet** [comment]
Same problem here. IP is not blacklisted as I can login manually through the website without any problems. 

Probably a sudden api change?

If I’d known these problems with Somfy earlier then I would probably have chosen another system. 

Thank you for this plugin!
We use it daily here in this household.

**Comment by chicochaleur29** [comment]
Same problem here too....

**Comment by Reckziegel182** [comment]
Guys this is what fixed it for me. Backup the config, remove the plugin and leave it be for couple of hours. The system will stop poking the server which keeps reseting the black list timer. After three hours i’ve reinstalled the plugin and it staryed working.

**Comment by jasper-seinhorst** [comment]
That means that we should increase the default polling time. We probably poll the API too much. I don't have an license, is the max requests per day documented?

**Comment by WTell** [comment]
Same issue here.

Logs:

```
hb-Tahoma-error.log last 15 lines:
0|hb-Tahom | [9/30/2020, 1:00:31 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:00:51 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:01:11 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:01:31 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:01:51 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:02:11 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:02:31 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:02:51 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:03:11 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:03:31 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:03:51 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:04:11 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:04:31 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:04:51 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.
0|hb-Tahom | [9/30/2020, 1:05:11 PM] [Tahoma] homebridge-tahoma is taking a long time to load and preventing Homebridge from starting.

hb-Tahoma-out.log last 15 lines:
0|hb-Tahom | [9/30/2020, 12:52:51 PM] ---
0|hb-Tahom | [9/30/2020, 12:52:51 PM] Loading 1 platforms...
0|hb-Tahom | [9/30/2020, 12:52:51 PM] [Tahoma] Initializing Tahoma platform...
0|hb-Tahom | [9/30/2020, 12:52:51 PM] [Tahoma] Fetch accessories...
0|hb-Tahom | [9/30/2020, 12:52:51 PM] [Tahoma] Login TaHoma server...
0|hb-Tahom | [9/30/2020, 12:52:51 PM] [Tahoma] Unable to refresh states (not logged in)
0|hb-Tahom | [9/30/2020, 12:53:21 PM] [Tahoma] Fetch accessories...
0|hb-Tahom | [9/30/2020, 12:53:21 PM] [Tahoma] Login TaHoma server...
0|hb-Tahom | [9/30/2020, 12:54:21 PM] [Tahoma] Fetch accessories...
0|hb-Tahom | [9/30/2020, 12:54:21 PM] [Tahoma] Login TaHoma server...
0|hb-Tahom | [9/30/2020, 12:56:22 PM] [Tahoma] Fetch accessories...
0|hb-Tahom | [9/30/2020, 12:56:22 PM] [Tahoma] Login TaHoma server...
0|hb-Tahom | [9/30/2020, 1:00:22 PM] [Tahoma] Fetch accessories...
0|hb-Tahom | [9/30/2020, 1:00:22 PM] [Tahoma] Login TaHoma server...
```

I'm shutting it down now and trying again in an hour.

**Comment by richie80** [comment]
Today I have had the same error. Polling period is set to 60
Can you advise? 
[10/26/2020, 7:08:12 PM] [My Tahoma Box] undefined
[10/26/2020, 7:08:12 PM] [My Tahoma Box] Login TaHoma server...
[10/26/2020, 7:08:12 PM] [My Tahoma Box] Login fail: Bad credentials
[10/26/2020, 7:08:12 PM] [My Tahoma Box] Error while registering listener
[10/26/2020, 7:08:42 PM] [My Tahoma Box] undefined
[10/26/2020, 7:08:42 PM] [My Tahoma Box] Login TaHoma server...
[10/26/2020, 7:08:42 PM] [My Tahoma Box] Login fail: Bad credentials
[10/26/2020, 7:08:42 PM] [My Tahoma Box] Error while registering listener
[10/26/2020, 7:09:12 PM] [My Tahoma Box] undefined
[10/26/2020, 7:09:12 PM] [My Tahoma Box] Login TaHoma server...
[10/26/2020, 7:09:12 PM] [My Tahoma Box] Login fail: Bad credentials
[10/26/2020, 7:09:12 PM] [My Tahoma Box] Error while registering listener
[10/26/2020, 7:09:42 PM] [My Tahoma Box] undefined
[10/26/2020, 7:09:42 PM] [My Tahoma Box] Login TaHoma server...
[10/26/2020, 7:09:42 PM] [My Tahoma Box] Login fail: Bad credential

**Comment by filibuster007** [comment]
Same problem here too.

**Comment by Supermarco333** [comment]
It took me a bit of time to figure this out but the `alwaysPoll` config is deprecated, in fact it isn't in the code anymore since the commit  3cf2b96394609d2d9f1dfee1a854d9a5795aad48. You should remove that from your config file.
Just to make this clear, if you trigger `pollingPeriod`: HomeBridge will fetch changes through the API but it won't fetch the accessories. It is the goal of `refreshPeriod`. That's why `refreshPeriod` should be set to a pretty high value.

**Device <-1-> API/Server <-2-> Plugin**
refreshPeriod => force API to refresh IO device states (link 1), and report it to plugin (link 2). Involves local radio communication.
pollingPeriod => get fresher device states from server (link 2). Allows more frequent request. 

In a nutshell, if you don't have sensors that push themselves their information to the API, you don't need  `pollingPeriod`. Keep the `refreshPeriod` to check discrepancies if you use IO buttons for instance.

**Comment by filibuster007** [comment]
Thanks for your explanation! Will make the changes.
